File: lakeSide-hotel-demo-server-master/src/main/java/com/dailycodework/lakesidehotel/model/Category.java
package com.dailycodework.lakesidehotel.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;
import java.util.ArrayList;
import java.util.List;

/**
 * @author Simpson Alfred
 */
@Entity
@Getter
@Setter
@AllArgsConstructor
public class Category {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private  Long id;
    private String type;

    @Column(columnDefinition = "TEXT")
    private String description;

    @OneToMany(mappedBy="category", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private List<Room> rooms;

    public Category() {
        this.rooms = new ArrayList<>();
    }

}

File: lakeSide-hotel-demo-client-main/src/model/Category.js
class Category {
    constructor(id = null, type = null, description = null, rooms = null) {
        this.id = id;
        this.type = type;
        this.description = description;
        this.rooms = rooms; // Mảng các đối tượng RoomModel
    }
}

export default Category;